Ruby mastermind: Update how to play guide from Wikipedia to wikiHow (#28233)
TheOdinProject/curriculum: ruby/object_oriented_programming_basics/project_mastermind.md

### Introduction

> If you've never played Mastermind, a game where you have to guess your opponent's secret code within a certain number of turns (like hangman with colored pegs), if you've never played or just need a refresher on how to play you can check out this guide on [wikiHow](https://www.wikihow.com/Play-Mastermind). Each turn you get some feedback about how good your guess was -- whether it was exactly correct or just the correct color but in the wrong space.

Reviewer: ```suggestion
If you've never played [Mastermind](https://www.wikihow.com/Play-Mastermind), it's a game where you have to guess your opponent's secret code within a certain number of turns (like hangman with colored pegs). Each turn you get some feedback about how good your guess was -- whether it was exactly correct or just the correct color but in the wrong space.
```

<div class="lesson-content__panel" markdown="1">

  1. Think about how you would set this problem up!
  1. Build the game assuming the computer randomly selects the secret colors and the human player must guess them.  Remember that you need to give the proper feedback on how good the guess was each turn!
  1. Now refactor your code to allow the human player to choose whether they want to be the creator of the secret code or the guesser.
  1. Build it out so that the computer will guess if you decide to choose your own secret colors.  You may choose to implement a computer strategy that follows the rules of the game or you can modify these rules.
  1. If you choose to modify the rules, you can provide the computer additional information about each guess.  For example, you can start by having the computer guess randomly, but keep the ones that match exactly.  You can add a little bit more intelligence to the computer player so that, if the computer has guessed the right color but the wrong position, its next guess will need to include that color somewhere.

>   1. If you want to follow the rules of the game, you'll need to research strategies for solving Mastermind, such as this [post](https://puzzling.stackexchange.com/questions/546/clever-ways-to-solve-mastermind).

Reviewer: ```suggestion
  1. If you want to follow the rules of the game, you'll need to research [strategies for solving Mastermind](https://puzzling.stackexchange.com/questions/546/clever-ways-to-solve-mastermind).
```
